server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -461,16 +462,19 @@ void validate(
             }
             throw new IllegalArgumentException(msg);
         } else  {
-            Set<String> settingsDependencies = setting.getSettingsDependencies(key);
+            Set<Setting<?>> settingsDependencies = setting.getSettingsDependencies(key);
```

Comment:
This change is made so that we can validate dependencies exist using the actual settings object, and can therefore use fallback settings when doing the validation, which would not be possible if using the top-level dependent key only.